### PR TITLE
[60] Connect user data to API

### PR DESCRIPTION
**Pull request by PKhing**
## What did you do

- Add mock login page
- Setup axios and fetch user data from API
- Edit Header to change according to login status (No Icon yet, let P'Gear do it)
image
image

- Add layout context for convenient header setting
  - to hide header call `useHideHeader()` in your page
  - to change focus call `useHeaderFocus('label')` in your page
- Implement test for all changed files

### CODE REVIEW

File: apps/api/src/main.ts [hunk 1]
```diff
@@ -11,6 +11,10 @@ async function bootstrap() {
   const app = await NestFactory.create(AppModule)
   app.useGlobalPipes(new ValidationPipe())
   app.use(cookieParser())
+  app.enableCors({
+    credentials: true,
+    origin: true,
```
Comment (palmcm) [inline apps/api/src/main.ts:16]: maybe this should be config

File: packages/database/src/mocks/getBaseMock/index.ts [hunk 2]
```diff
@@ -48,7 +48,8 @@ export const getBaseMock = (
       return {
         userId: index,
         email: faker.helpers.unique(faker.internet.email),
-        password: faker.internet.password(),
+        password:
+          '$2a$10$jikY3o7Apw/.NMTQjtLC1OMLUwEM73dpWaM7T5WEmjB5lguG/wZce',
```
Comment (palmcm) [inline packages/database/src/mocks/getBaseMock/index.ts:52]: what is password for this

Comment (PKhing) [reply packages/database/src/mocks/getBaseMock/index.ts:52]: 'password' :joy:

**Review by miello — APPROVED** [review]
I think that `Header` components should name `Navbar` instead

**Pull request merged**